torch/_higher_order_ops/utils.py

```diff
@@ -124,13 +124,16 @@ def wrapped(*args):
 @contextmanager
 def _set_compilation_env():
     _old_is_tracing = torch.fx._symbolic_trace._is_fx_tracing_flag
+    _old_allow_empty_graphs = torch._dynamo.config.allow_empty_graphs
```

Comment:
What is this?

Reply:
The issue is tracked here https://github.com/pytorch/pytorch/issues/144360: when dynamo finds the top-level frame produces no graph, the current behavior is to fallback to eager then when it encounters an inner call to another function, it will try to inspect that function again (which is unnecessary). During inspect the inner call, we trace into the python dispathcer logic in _ops.py which is not tracable as of today. So the proper fix can be either 1. allow dispatch logic to be tracable or 2. fixing 144360. but it exposes some bugs in existing tests so we have to have a temporary flag to control the behavior, which essentially allows dynamo to store an empty graph for a frame without falling back to eager, which is what we want.